registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
These examples are just canonicalizing leading `/`.

> i waned to use the actual path itself and the actual paths

I'm confused by this: the storage driver maps a simplified VFS to a `/` rooted path space. The path is just a name for a resource. It doesn't support anything close to "actual paths", as the backends don't support this. For example, relative paths aren't supported. The fact that the front end and the backend have similar paths is just an implementation detail. You could map `/foo` to `10` on the backend, you wanted. In general, the backend should make the path look as needed to store data.

The goal of this path space is to define a path set that will work on a large number of systems. Opening it up to `:`, for instance, breaks windows compatibility.

I am not sure we would have merged this or the previous PR upstream.

Do you have an example of how you're using this?